requirements/dev.txt

```diff
@@ -0,0 +1,18 @@
+# Development "requirements".  These are tools that are extremely useful or
+# even required to run the test suite, but not at all required by Zino at
+# runtime:
+black
+build
+coverage
+ipython
+isort
+pre-commit
+pytest
+pytest-asyncio<0.22.0
+pytest-timeout
+retry
+ruff
+snmpsim>=1.0
```

Comment:
```suggestion
snmpsim>=1.0,!=1.1.6
```

See #370

Reply:
Strange, I copied the list from `pyproject.toml`, and that restriction was there at the time. No idea why it disappeared...

Reply:
Nope, #370 only updated `tox.ini`, not `pyproject.toml`.  But I just rebased this PR and added a commit to include the same constraint in `dev.txt`